docs/development/pigs/pig-010.rst

```diff
@@ -0,0 +1,304 @@
+
+This requires a spherical point in polygon algorithm, which probably is hard to
+implement. Discussion on this point is in `astropy-regions GH 46`_.
+
+- Implement ``regions.CircleSkyRegion.contains`` (see above)
+- Implement ``regions.PolygonSkyRegion.contains``, also use for rectangle? (see TODO)
+
+Merge `astropy-regions GH 221` to make it easier to plot sky regions? Is it a good idea or will it cause problems with "real" sky regions?
+
+Masks
+-----
+
+, via the existing
+`gammapy.maps.WcsGeom.region_mask`_ method, and then only the pixel mask is
+used.
+
+- Use bool or int as `dtype`?
+- Define meaning of 0 and 1 (probably use as-is)
+
+Sometimes binary masks are not enough. E.g. in ``scikit-image`` or
+``scipy.ndimage`` often "label images" are used, in cases where there are more
+than two disjoint regions ("in" and "out" as in a binary mask), e.g. 
+
+Image analysis
+--------------
+
+In `gammapy.image`_ there's currently a convenience functions and classes for 2D
+image analysis that aren't used anywhere in Gammapy. They could be removed,
+replaced with an example notebook how to do region-based with ``scipy.ndimage``
+and ``scikit-image`` and `photutils`_.
+
+Some are region-related, e.g. these:
+
+- https://docs.gammapy.org/0.11/api/gammapy.image.measure_containment.html
+- https://docs.gammapy.org/0.11/api/gammapy.image.ImageProfileEstimator.html
+- https://docs.gammapy.org/0.11/api/gammapy.image.ImageProfile.html
+
+Another commonly needed feature in gamma-ray astronomy is radial profiles:
+https://github.com/gammapy/gammapy/pull/2129#issuecomment-493415766
+Actually we had those at one point in Gammapy:
+https://github.com/gammapy/gammapy/blob/v0.7/gammapy/image/radial_profile.py
+
+TODO: discuss what to keep and write it down here.
+
+Christoph: I'd prefer to replace most of ``gammapy.image`` with documentation.
+This is stuff users can script with a few lines, and trying to pre-script it in
+Gammapy to be useful to many users is very hard. We're not doing users or us a
+favour to keep this in Gammapy at this point.
+
+Tables
+------
+
+Currently we have as examples `gammapy.data.EventList`_,
+`gammapy.data.ObservationTable`_ and `gammapy.catalog.SourceCatalog`_ which each
+are a wrapper or subclass of an `astropy.table.Table`. 
+`select_sky_box`_
+`select_sky_circle`_
+`skycoord_from_table`_
+
+Re-exposed on EventList and ObservationTable methods
+
+Replace with ``select_region`` or remove completely?
+
+Offer two methods to make selection masks, and also to apply the selection?
+
+TODO: aligned lon / lat sky box - needs extra region class?
+
+Rotated regions
+---------------
+
+- One more the most complex pieces of code in Gammapy:
+  https://github.com/gammapy/gammapy/blob/master/gammapy/background/reflected.py
+- See https://github.com/gammapy/gammapy/pull/2089
+- What to do?
+
+Rotate for polygons was already implemented (see `astropy-regions GH 91`_),
+still there, even though it's currently unused.
+
+Region strings
```

Comment:
I looked around a bit today concerning region strings.

The problem is that there's different region string systems in use.
- http://ds9.si.edu/doc/ref/region.html
- http://www.ivoa.net/documents/ADQL/20180112/PR-ADQL-2.1-20180112.html
- https://heasarc.gsfc.nasa.gov/docs/software/fitsio/quick/node23.html

And there's the question not just how to specify a region via a string as a keyword argument, or from a config file.
Probably we'd want to use the same format to store the region in meta, and there we have yet another one already:
https://gamma-astro-data-formats.readthedocs.io/en/latest/skymaps/healpix/index.html#healpix-region-string

So while I still like the idea to allow region strings in Gammapy, I would suggest to not add support for it now, and to not work out a solution in this PIG. Just mention it in the "future work" section at the end fo the PIG.

@registerrier @adonath - OK?

Reply:
It is clear to me that we only support string serialisation which are supported in `astropy.regions`. So I think the work should happen there. So e.g. introduce a convenience factory method such as `SkyRegion.from_ds9_string("")`. Note that this technically already works:
```
from regions import DS9Parser
reg_string = 'galactic \n circle(42,43,3)'
parser = DS9Parser(reg_string)
parser.shapes[0].to_region()
```
So I think there is nothing to be done from the Gammapy side...

Reply:
I added a long new section, actually proposing to add DS9 region string support. Please have a look.